examples/shadow/shader.wgsl

```diff
@@ -0,0 +1,114 @@
+[[location(0)]]
+var<in> in_position: vec4<i32>;
+[[builtin(position)]]
+var<out> out_position: vec4<f32>;
+
+[[block]]
+struct Globals {
+    view_proj: mat4x4<f32>;
+    num_lights: vec4<u32>;
+};
+
+[[group(0), binding(0)]]
+var<uniform> u_globals: Globals;
+
+[[block]]
+struct Entity {
+    world: mat4x4<f32>;
+    color: vec4<f32>;
+};
+
+[[group(1), binding(0)]]
+var<uniform> u_entity: Entity;
+
+[[stage(vertex)]]
+fn vs_bake() {
+    out_position = u_globals.view_proj * u_entity.world * vec4<f32>(in_position);
+}
+
+
+[[location(1)]]
+var<in> in_normal: vec4<i32>;
+
+[[location(0)]]
+var<out> out_normal_vs: vec3<f32>;
+[[location(1)]]
+var<out> out_position_vs: vec4<f32>;
+
+[[stage(vertex)]]
+fn vs_main() {
+    const w: mat4x4<f32> = u_entity.world;
+    out_normal_vs = mat3x3<f32>(w.x.xyz, w.y.xyz, w.z.xyz) * vec3<f32>(in_normal.xyz);
+    out_position_vs = u_entity.world * vec4<f32>(in_position);
+    out_position = u_globals.view_proj * out_position_vs;
+}
+
+// fragment shader
+
+[[block]]
```

Comment:
FYI. It's invalid to declare an array of block-decorated struct.
WGSL doesn't have array-of-buffers, but Vulkan SPIR-V does

Reply:
Right. At the time this was written, our `[[block]]` semantics was a bit different. Today, there is no `[[block]]` on it https://github.com/gfx-rs/wgpu-rs/blob/69be1ff88a073b3a7f9470044834e75c22f7b4ad/examples/shadow/shader.wgsl#L54